### PR TITLE
feat: add support for o1 model benchmark and suite

### PR DESCRIPTION
**Pull request by daiwaid**
#### What this PR does / why we need it:
- support for o1 model benchmark and suite
- optional rate limiter to limit match requests per minute
- add `--rpm` flag to `benchmark.run`

#### Which issue(s) this PR addresses (if any):

Addresses #

#### Does this PR introduce a user-facing change?

```release-note
NONE
```

### CODE REVIEW

File: libem/match/parameter.py [hunk 1]
```diff
@@ -19,6 +19,16 @@
     },
 )
 
+# optional requests per minute limit
+rpm = Parameter(
+    default=0
+)
+
+# enable system prompt
```
Comment (zenodflow) [inline libem/match/parameter.py:27]: this should be in the libem.parameter.py and also use Index to encode this conditional knowledge

File: libem/match/function.py [hunk 2]
```diff
@@ -208,7 +208,7 @@ async def batch(left: list[str], right: list[str]) -> list[dict]:
         )])
 
     _prompt = [
-        {"role": "system", "content": system_prompt},
+        {"role": "system" if parameter.system_prompt() else "user", "content": system_prompt},
```
Comment (zenodflow) [inline libem/match/function.py:211]: pull this knowledge to the libem.parameter too, make this system_role

File: libem/core/exec.py [hunk 3]
```diff
@@ -1,21 +1,32 @@
...
 from tqdm.asyncio import tqdm
+from aiolimiter import AsyncLimiter
 
 import libem
 
 
 async def proc_async_tasks(
         tasks: List[Coroutine],
         max_async_tasks: int = libem.LIBEM_MAX_ASYNC_TASKS,
+        rpm: int = -1,
```
Comment (zenodflow) [inline libem/core/exec.py:12]: make rpm consistently -1 as default

**Comment by daiwaid** [comment]
Addressed all comments.

File: libem/core/exec.py [hunk 4]
```diff
@@ -1,21 +1,32 @@
...
         desc: str = "Processing Tasks") -> List:
-    ''' Run async tasks under na imposed max concurrent tasks. '''
+    ''' 
+        Run async tasks under na imposed max concurrent tasks 
+        with an optional rate limiter.
+    '''
 
     sem = asyncio.Semaphore(max_async_tasks)
+    if rpm > 0:
+        limiter = AsyncLimiter(rpm)
```
Comment (zenodflow) [inline libem/core/exec.py:21]: nit:

else:

limiter = None

File: libem/core/exec.py [hunk 5]
```diff
@@ -1,21 +1,32 @@
...
 
     async def _run(sem, task):
         async with sem:
-            return await task
+            if rpm > 0:
```
Comment (zenodflow) [inline libem/core/exec.py:25]: if limiter:

File: libem/match/parameter.py [hunk 6]
```diff
@@ -7,6 +7,7 @@
 # model call configurations
 model = libem.parameter.model
 temperature = libem.parameter.temperature
+system_role = libem.parameter.system_role
```
Comment (zenodflow) [inline libem/match/parameter.py:10]: space after

File: libem/optimize/cost/openai.py [hunk 7]
```diff
@@ -1,6 +1,10 @@
 from libem.optimize.cost import cache
 
 model_info = None
+null_model_info = {
```
Comment (zenodflow) [inline libem/optimize/cost/openai.py:4]: is there a reason why we don't just handle the null case within the get_model_info call?

**Review by zenodflow — APPROVED** [review]
lgtm - nice work! Just a few minor comments

**Comment by daiwaid** [comment]
Done.

**Comment by zenodflow** [comment]
@daiwaid Nice. Btw., in future, see if you can squash merge for single feature PR so that intermediate commits such as "stylistic fixes" won't enter commit history.

**Pull request merged**